Build: Switch from UglifyJS to SWC minify, make the minified file ES5

mgol opened the pull request:
More recent UglifyJS versions have started converting regular functions to arrow ones, making ES5 source file migrated to a ES2015+ minified one. We want to avoid that even in 1.14.x as long as we keep the source file in ES5. But the bigger problem is that we generate ES2015+ minified code even for 1.13.3 releases which support IE.

Fixes gh-629
Ref mishoo/UglifyJS#5967
Ref jquery/download.jqueryui.com#629

timmywil reviewed (approved):
Looks good, but we talked about verifying es5 at runtime

mgol commented:
@timmywil I had to update GitHub Actions as the older ones didn't work anymore; I copied automatic Dependabot updates setup from Core as well.

I reverted the minifier change to make sure new tests fail and they do, with expected messages complaining about parsing errors:
https://github.com/jquery/download.jqueryui.com/actions/runs/13638505216/job/38122855881?pr=630

I reverted the reverts and the PR is green now.

fnagel reviewed (approved):
+1 by reading
